dna/indexcontent/indexcontent.js

```diff
@@ -0,0 +1,205 @@
+function genesis(){
+
+  //Calling addAnchor function for creating the base anchor
+  baseAnchorHash = call("anchor","addAnchor","");
+  debug("Base anchor added with hash - "+baseAnchorHash);
+
+  var ContentToIndex1 = {content:"holodex : We are Indexing this content using holodex app. this",details:"can include timestamp, etc."};
+  ContentToIndexhash1 = makeHash(ContentToIndex1);
+
+  var ContentToIndex2 = {content:"holodex can also be used for searching keywords",details:"can include timestamp,lication, etc."};
+  ContentToIndexhash2 = makeHash(ContentToIndex2);
+
+
+  //called in genesis temporarily. Once bridging between apps gets workig, index content will be called from the HC app that is
+  //using holodex
+  IndexContent(ContentToIndex1.content,ContentToIndexhash1,"English");
+  IndexContent(ContentToIndex2.content,ContentToIndexhash2,"English");
+
+}
+
+//Function called by the HC app to search for a string of words and get all the objects indexed for the words.
+function searchKeywords(searchString)
+{
+  var searchArr = searchString.split(/,| |:|-/);
+  var i = searchArr.length;
+  var mergedList = [];
+  var list = [];
+
+  debug("Array contents : ");
+  i--;
+
+  while(i>=0)
+  {
+    debug("While loop ------------ i = "+i);
+    debug(searchArr[i]);
+    list = call("anchor","anchor_list",searchArr[i]);
+
+    var temp = JSON.parse(list);
+    debug(list);
+    for(var m=0;m<temp.length;m++)
+    {
+      var temp1 = JSON.parse(temp[m].Anchor_Text);
+      debug(temp1.Anchor_Text);
+      mergedList=union(mergedList,temp1.Anchor_Text);
+    }
+
+    i--;
+  }
+  var jsonmer = JSON.parse(JSON.stringify(mergedList));
+
+  return jsonmer;
+}
+
+//To provide a list of all the objects that are indexed against the search string
+function union(mergedList,list)
+{
+
+  if(mergedList.length==0)
+  {
+    if(list != "")
+      mergedList.push(list);
+  }
+  else
+  {
+    check=find(mergedList,list);
+
+    if(check==false)
+    {
+      if(list != "")
+      mergedList.push(list);
+    }
+  }
+  debug(mergedList);
+  return mergedList;
+}
+
+//Function to check if the object has already been covered in the merged list
+function find(mainArr, check)
+{
+
+  for(i=0;i<=mainArr.length;i++)
+  {
+
+    if(mainArr[i] == check)
+    {
+      return true;
+    }
+  }
+  return false;
+}
+
+//Index content function is called from a HC application by passing the content and the hash of the object so that the link cant be
+//made directly to the object.
+function IndexContent(content,hashOfObject,language)
+{
+
+
+  var HTIgnoreWords = getIgnoreWords(language);
+  //debug(HTIgnoreWords);
+
+  var IgnoreWords = "this is the a an are and can also with : -";
```

Comment:
cleanup, you can remove this now that you have getIngnoreWords,   or perhaps you could use some code like it in getIngoreWords to load the hash, so you don't have to manually set each one...